### DISCUSSION TITLE
Returning Row

### DISCUSSION THREAD

**Discussion by zuhairthabit01** [SeaQL/sea-orm#1776] [Q&A]
Currently, my update function is returning the entire row of data. However, I want it to return only the updated columns. Here is the function in question

```
//update full_name
pub async fn update_user_full_name(db: &DatabaseConnection, user_id: i32, new_full_name: String) -> Result<users::ActiveModel, DbErr> {
    let existing_user = Users::find_by_id(user_id).one(db).await?;
    match existing_user {
        Some(user) => {
            let utc_now = Utc::now();
            let updated_at = utc_now.naive_utc();
            let mut updated_user = user.into_active_model();
            updated_user.full_name = ActiveValue::set(Option::from(new_full_name));
            updated_user.updated_at = ActiveValue::set(updated_at);
            updated_user.clone().save(db).await?;
            Ok(updated_user)
        }
        None => Err(DbErr::Custom(“User ID does not exist”.into()))
    }
}
```

[2023-07-23T21:44:16Z INFO  sqlx::query] UPDATE "users" SET "full_name" …; rows affected: 0, rows returned: 1, elapsed: 432.299ms
    
    UPDATE
      "users"
    SET
      "full_name" = $1,
      "updated_at" = $2
    WHERE
      "users"."id" = $3 RETURNING "id",
      "user_id",
      "full_name",
      "email",
      "telegram_id",
      "email_push",
      "telegram_push",
      "web_push",
      "skip_email",
      "skip_telegram",
      "skip_web",
      "skip_reason",
      "created_at",
      "updated_at"

**Comment by billy1624** [comment]
Hey @zuhairthabit, I think it's hard for SeaORM to determine the exact list of columns need `RETURNING`. Consider this, one can have a database trigger to update the value of a column on update.
